### PR TITLE
Added signalfx collectd plugin build and upload scripts

### PR DESCRIPTION
**Pull request by udaysagar2177**

### CODE REVIEW

File: build-plugin/debian/copyright [hunk 1]
```diff
@@ -0,0 +1,29 @@
+Format: http://www.debian.org/doc/packaging-manuals/copyright-format/1.0/
+Upstream-Name: signalfx-collectd-plugin
+Source: <url://example.com>
+
+Files: *
+Copyright: 2015 SignalFx, Inc. <support+deb@signalfx.com>
+License: GPL-3.0+
```
Comment (tedoc2000) [inline build-plugin/debian/copyright]: I don't think we are GPL-3.0+ lets talk with jack to figure out what we must be..

File: build-plugin/debian/rules [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/make -f
+%:
+	dh $@
+override_dh_install:
+	dh_install signalfx_metadata.py  opt/signalfx-collectd-plugin
```
Comment (tedoc2000) [inline build-plugin/debian/rules]: I think we are going to end up with a whole directory of files.. lets figure out how to deal with that.

File: build-plugin/sfx_scripts/jenkins-build [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+if ! [ -z $JOB_NAME ] && ! [ -z $BASE_DIR ]; then
+  trap "docker rm ${JOB_NAME}" EXIT
+  #Build the packages
+  docker run --privileged --name ${JOB_NAME} -t -v ${BASE_DIR}/${JOB_NAME}/signalfx-collectd-plugin:/opt/signalfx-collectd-plugin -v ${BASE_DIR}/${JOB_NAME}/collectd-build-ubuntu/build-plugin:/opt/collectd-plugin-build-debian -v ${BASE_DIR}/${JOB_NAME}/workspace:/opt/workspace -v ${BASE_DIR}/${JOB_NAME}/result:/opt/result quay.io/signalfuse/collectd-build-ubuntu /opt/collectd-plugin-build-debian/sfx_scripts/cmdseq
+
+  if [ "${BUILD_PUBLISH}" = "True" ] ; then
+        # Upload the packages to Amazon S3 bucket
+        aws s3 rm --recursive s3://collectd-builds-ubuntu/signalfx-collectd-plugin/
+        aws s3 cp --recursive ${BASE_DIR}/${JOB_NAME}/result/ s3://collectd-builds-ubuntu/signalfx-collectd-plugin/
```
Comment (tedoc2000) [inline build-plugin/sfx_scripts/jenkins-build]: I think you should use a separate directory (tempdir) like wentao does in his jobs..
Either way you don't want to use BASE_DIR.. that is the PATH outside the jenkins docker image.. you just want /opt/jenkins/workspace/${JOB_NAME}/result/

**Comment by mpetazzoni** [comment]
That's a whole lot of duplicated files here (the entirety of the `sfx_scripts` and `build-plugin/sfx_scripts` directories for example). What's going on?

**Comment by udaysagar2177** [comment]
we want the repo to hold the build files of both collectd and signalfx
collectd plugin files.

**Pull request merged**